src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PETypeParameterSymbol.cs

```diff
@@ -212,7 +212,7 @@ private ImmutableArray<TypeSymbolWithAnnotations> GetDeclaredConstraintTypes()
                             continue;
                         }
 
-                        var type = TypeSymbolWithAnnotations.Create(typeSymbol, isNullableIfReferenceType: null);
+                        var type = TypeSymbolWithAnnotations.CreateNonNull(containingType.NonNullTypes, typeSymbol);
```

Comment:
Feels like we should either use the containingMethod or the containingType, for getting the NonNullTypes context. Depending if the type parameter belongs to a method or a type.

Reply:
Good catch. I've fixed this, but I believe we have other blocking issues that prevent the change from being observable. For now, I've added a `// PROTOTYPE` comment to test the difference.

---
In reply to: [203875795](https://github.com/dotnet/roslyn/pull/28687#discussion_r203875795) [](ancestors = 203875795)